src/GitVersionCore/BuildAgents/CodeBuild.cs

```diff
@@ -20,7 +21,7 @@ public void WithPropertyFile(string propertiesFileName)
             file = propertiesFileName;
         }
 
-        protected override string EnvironmentVariable { get; } = EnvironmentVariableName;
+        protected override string EnvironmentVariable => throw new InvalidOperationException();
```

Comment:
Can you please provide the exception with a message explaining why accessing `EnvironmentVariable` is an invalid operation?

Reply:
I have updated the PR based on your suggestion.  While considering an appropriate error message, I came to conclusion that an NotSupportedException may be clearer exception to use.